webapp/src/component/activity/types/getTextDiff.tsx

```diff
@@ -0,0 +1,49 @@
+import { styled } from '@mui/material';
+import clsx from 'clsx';
+import { diffWordsWithSpace } from 'diff';
+
+import { DiffValue } from '../types';
+
+const StyledDiff = styled('span')`
+  & .removed {
+    color: ${({ theme }) => theme.palette.activity.removed};
+    text-decoration: line-through;
+  }
+  & .added {
+    background: ${({ theme }) => theme.palette.activity.addedHighlight};
+  }
+`;
+
+export const getTextDiff = (input?: DiffValue) => {
+  const oldInput = input?.old;
+  const newInput = input?.new;
+  if (oldInput && newInput) {
+    const diffed = diffWordsWithSpace(oldInput, newInput);
+    return (
+      <StyledDiff>
+        {diffed.map((part, i) => {
+          return (
+            <span
+              key={i}
+              className={clsx({ added: part.added, removed: part.removed })}
```

Comment:
Here it's also not clear, that the span may have only one of the classes.

Reply:
👍